nemesis/arenaPrep.py

```diff
@@ -1121,8 +1118,8 @@ def kilika_final_shop():
     menu.auto_sort_equipment()
 
     gilNeeded = 3500000 - memory.main.get_gil_value()
-    weaponBuys = int(gilNeeded / 26150)
-    weaponBuys += 1  # for safety
+    # Get minimum needed, plus one for safety. Max 99 total.
+    weaponBuys = min(int(gilNeeded / 26150), 98) + 1
```

Comment:
This was a fix pitched by @coderwilson and it seemed to work on my machine at least once.

Reply:
I'm editing this further for an "out of gil" scenario, but this should be fine to merge.